duckduckgo/duckduckgo-autofill
# l10n: add basic localization support to device abstraction

**Reviewer:** @shakyShane 
**Asana:** https://app.asana.com/0/72649045549333/1206933877192895

## Description
(Reimplements #540 in a mergeable state 😄)

This is a series of four commits (each individually buildable and testable, ready for a rebase merge or merge commit) that add support for localization to the autofill HTML UI. Check each commit message for more details, but the TL;DR is:

1. Add `.userPreferences.language` to runtime configuration
2. Implement basic translation and wire it up to the Device abstraction
3. Copy the already-provided language from the extension's runtime config
4. Translate two strings as a small demo

## Steps to test
1. The usual `npm run verify:local` on each commit
2. On the final commit, test end-to-end with the `xa` pseudolanguage:
    1. `npm run build` in this repo
    2. Clone https://github.com/duckduckgo/duckduckgo-privacy-extension next to this repo if you haven't yet
    3. Apply [this patch](https://gist.github.com/sjbarag/bbeec6b10c3e89d1b48e5ba70af75c1d)
    4. `npm install && npm link ../duckduckgo-autofill && npm run dev-chrome` in that repo
    5. Launch Google Chrome and load a signup page
    6. You should see translated (in a pseudolocale) text for the "Use personal duck address" button
